### PR TITLE
Refactor data loading for the top function

### PR DESCRIPTION
**Pull request by gabrielbussolo**
# Summary of this proposal

This PR adds additional error handling and refactors the `loadData` function in `top_project.go` to make it more modular and testable. Specifically, it extracts the data loading logic into three new functions: `loadProjectData`, `loadAgentMetrics`, and `loadAgents`.

This change is expected to improve the maintainability of the codebase and make it easier to add new features in the future.

## Issue(s) number(s)

*Disclaimer: Please do not create a Pull Request without creating an issue first.*

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

**Review by jasmingacic — CHANGES_REQUESTED** [review]
While the code looks good. Can you please provide me with context why we need this?

**Comment by gabrielbussolo** [comment]
> While the code looks good. Can you please provide me with context why we need this?

tl;dr; readability 

its described on the PR, but this work was triggered by https://github.com/calyptia/cli/issues/297

**Review by jasmingacic — DISMISSED** [review]

**Comment by niedbalski** [comment]
@nicolasparada can you review this one?

**Review by nicolasparada — DISMISSED** [review]
The imports are not sorted, but should be good to merge any way.

**Comment by nicolasparada** [comment]
Fixed the import sorting with

```bash
goimports -w -local github.com/calyptia/ .
```
